### PR TITLE
{bio}[ictce/5.4.0,goolf/1.4.10] MIGRATE 3.4.4/3.6.6 (REVIEW)

### PR DESCRIPTION
**Pull request by jordiblasco**

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
@jordiblasco: changed the PR title to make it fit the policy proposal at https://github.com/hpcugent/easybuild/wiki/Policy-for-easyconfig-pull-requests

@wpoely86 or @pescobar or @fgeorgatos: care to tackle reviewing/testing this until it's ready to merge?

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# 
+# This is an easyconfig file for EasyBuild, see https://github.com/hpcugent/easybuild
+# Authors::   Jordi Blasco <jordi.blasco@nesi.org.nz>
+# New Zealand eScience Infrastructure (NeSI)
+
+name = 'MIGRATE'
+version = "3.4.4" 
+
+homepage = 'http://popgen.sc.fsu.edu/Migrate/Migrate-n.html'
+description = """Migrate estimates effective population sizes and past migration rates between n population assuming a migration matrix model with asymmetric migration rates and different subpopulation sizes. Migrate uses maximum likelihood or Bayesian inference to jointly estimate all parameters."""
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb]: wrap this line.

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+binary_files = ["migrate-n-mpi", "migrate-n"]
+
+# The sanity test MUST be tuned before going production and submitting your contribution to upstream git repositories
+sanity_check_paths = {
+    'files': ['bin/%s' % x for x in binary_files],
+    'dirs': []
+}
+
+# You SHOULD change the following line; Kindly consult other easyconfigs for possible options
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb]: drop this comment :wink:

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb]: has this easyconfig a special easyblock? Else an `easyblock = ` line should be added.

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+homepage = 'http://popgen.sc.fsu.edu/Migrate/Migrate-n.html'
+description = """Migrate estimates effective population sizes and past migration rates between n population assuming a migration matrix model with asymmetric migration rates and different subpopulation sizes. Migrate uses maximum likelihood or Bayesian inference to jointly estimate all parameters."""
+
+toolchain = {'name': 'ictce', 'version': '5.4.0'} 
+toolchainopts = {'opt': True, 'optarch': True, 'unroll': True, 'static': False}
+
+# For sources line to work correctly with --try-software-version, you MUST employ %s OR use a construct like SOURCE_TAR_GZ
+oldversions="oldversions/%s.x/%s/" % (".".join(version.split('.')[:1]), ".".join(version.split('.')[:2])) 
+sources = ['%(namelower)s-%(version)s.src.tar.gz']
+source_urls = ['http://popgen.sc.fsu.edu/newversions/', 'http://popgen.sc.fsu.edu/currentversions/', 'http://popgen.sc.fsu.edu/%s' % oldversions ]
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb]: put every url on a new line

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+name = 'MIGRATE'
+version = "3.4.4" 
+
+homepage = 'http://popgen.sc.fsu.edu/Migrate/Migrate-n.html'
+description = """Migrate estimates effective population sizes and past migration rates between n population assuming a migration matrix model with asymmetric migration rates and different subpopulation sizes. Migrate uses maximum likelihood or Bayesian inference to jointly estimate all parameters."""
+
+toolchain = {'name': 'ictce', 'version': '5.4.0'} 
+toolchainopts = {'opt': True, 'optarch': True, 'unroll': True, 'static': False}
+
+# For sources line to work correctly with --try-software-version, you MUST employ %s OR use a construct like SOURCE_TAR_GZ
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb]: drop?

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+oldversions="oldversions/%s.x/%s/" % (".".join(version.split('.')[:1]), ".".join(version.split('.')[:2])) 
+sources = ['%(namelower)s-%(version)s.src.tar.gz']
+source_urls = ['http://popgen.sc.fsu.edu/newversions/', 'http://popgen.sc.fsu.edu/currentversions/', 'http://popgen.sc.fsu.edu/%s' % oldversions ]
+
+patches = []
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb]: drop if emtpy

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+prebuildopts = 'cd ./src && '
+
+buildopts = ' mpis && make cleanall && make thread '
+
+preinstallopts = 'cd ./src && mkdir -p %(installdir)s/{bin,man/man1} && '
+
+binary_files = ["migrate-n-mpi", "migrate-n"]
+
+# The sanity test MUST be tuned before going production and submitting your contribution to upstream git repositories
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb]: drop?

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+patches = []
+
+dependencies = [('zlib', '1.2.5')]
+
+preconfigopts = 'cd ./src && sed -i "s/ \-fast / /g" configure && ' 
+
+configopts = '--with-zlib=system --enable-sse2 '
+
+parallel = 1
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-ictce-5.4.0.eb]: put this below all the `*opts` lines?

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by fgeorgatos** [comment]
@jordiblasco : cosmetic changes needed here

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
missing `easyblock = 'ConfigureMake'` specs

**Comment by valtandor** [comment]
@jordiblasco this is your old MIGRATE PR

**Comment by migueldiascosta** [comment]
@jordiblasco, this PR is being closed for the following reason(s): no activity for > 6 months.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**